src/tool_operate.c

```diff
@@ -1521,27 +1769,30 @@ static CURLcode single_transfer(struct GlobalConfig *global,
         my_setopt_str(curl, CURLOPT_KEYPASSWD, config->key_passwd);
         my_setopt_str(curl, CURLOPT_PROXY_KEYPASSWD, config->proxy_key_passwd);
 
-        if(use_proto == proto_scp || use_proto == proto_sftp) {
-
-          /* SSH and SSL private key uses same command-line option */
-          /* new in libcurl 7.16.1 */
-          my_setopt_str(curl, CURLOPT_SSH_PRIVATE_KEYFILE, config->key);
-          /* new in libcurl 7.16.1 */
-          my_setopt_str(curl, CURLOPT_SSH_PUBLIC_KEYFILE, config->pubkey);
-
-          /* new in libcurl 7.17.1: SSH host key md5 checking allows us
-             to fail if we are not talking to who we think we should */
-          my_setopt_str(curl, CURLOPT_SSH_HOST_PUBLIC_KEY_MD5,
-                        config->hostpubmd5);
-
-          /* new in libcurl 7.80.0: SSH host key sha256 checking allows us
-             to fail if we are not talking to who we think we should */
-          my_setopt_str(curl, CURLOPT_SSH_HOST_PUBLIC_KEY_SHA256,
-              config->hostpubsha256);
-
-          /* new in libcurl 7.56.0 */
-          if(config->ssh_compression)
-            my_setopt(curl, CURLOPT_SSH_COMPRESSION, 1L);
+        if(proto) {
+          if(Tool_safecmp(proto, (char *) proto_scp) ||
+              Tool_safecmp(proto, (char *) proto_sftp)) {
```

Comment:
Again you first check `proto` so there is no need for "safe" checks.

Reply:
Yeah, that was my assumption too. That turned out false.
There are some build nodes in one of the many previous buids from today that just fail on that. Believe me, i'm not making these changes for fun anymore.